### ISSUE TITLE
Project is not installed correctly during tox or travis run, it's not possible to run unit tests via tox

### ISSUE THREAD

**Issue by mbukatov** [Tendrl/alerting#24]
There are multiple issues with current tox and travis configuration:

* `tox.ini` file installs dependencies from requirements file directly, see line `deps = -r{toxinidir}/requirements.txt` (for full description, see https://github.com/Tendrl/alerting/issues/18#issuecomment-270661999)
* `tox.ini` file includes installation of dev packages for debian, which fails on fedora - I would rather include this in `.travis.yml` so that Travic CI run still works, but I'm able to run `tox` on fedora and under nonroot account as well
* Travis CI job runs only `pep8` and `cover` enviroments, no unit tests are actually executed (see `script:` section in `.travis.yml` (note that there are no unit tests at this point, but it would be better to have the setup done from the start so that developers are actually encouraged to write them during development)
* the project is not installed from sdist, see `skipsdist = True` in `tox.ini`
* I'm not sure if this line in `tox.ini` is correct: `install_command = pip install -U {opts} {packages}`

I may append items into this list when I fix some of the issues listed there, because I can't run it in the current state.

## Expected state

Running `tox` in the root directory of the project, executes all unit tests and other checks, assuming I have all dev packages installed, both on Fedora/CentOS and Debian/Ubuntu. This would encourage developers to actually write unit tests, because there would be an easy and direct way to execute them.

Travis CI job would execute unit tests during it's run via tox, along other checks it does right now.

## Related Issues

The same kind of issues are applicable to tox and travis configuration of all other python Tednrl projects. I have opened just this one now, and I would suggest to fix the other project based on resolution reached here.

**Comment by mbukatov** [comment]
Hmm, [when I tried to move apt install into travis config file](https://github.com/mbukatov/alerting/commit/ef85404d5d23e17f4ff5f93d43d3d09deb570ec0) (as suggested by [Travis CI documentation](https://docs.travis-ci.com/user/installing-dependencies/)), I noticed that it doesn't work:

```
$ sudo apt-get install -y build-essential libsystemd-journal-dev libsystemd-daemon-dev libsystemd-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package libsystemd-journal-dev
E: Unable to locate package libsystemd-daemon-dev
E: Unable to locate package libsystemd-dev

The command "sudo apt-get install -y build-essential libsystemd-journal-dev libsystemd-daemon-dev libsystemd-dev" failed and exited with 100 during .
```

See full Travis CI log: https://travis-ci.org/mbukatov/alerting/builds/189543125

It looks as if these lines were copied over from another Tendrl project, but it was actually never used.

**Comment by mbukatov** [comment]
usmqelabel: foo123

**Comment by mbukatov** [comment]
@Tendrl/qe

**Comment by mbukatov** [comment]
I'm going to propose a change of current CI setup, work in progress can be found in [my fix_ci temporary branch](https://github.com/mbukatov/alerting/commits/fix_ci).

**Comment by mbukatov** [comment]
After looking into details here, here is an updated summary of issues with the current setup:

* tox installs dependencies from `requirements.txt` directly (to understand why this is bad, see Tendrl/alerting#18)
* ubuntu dev setup for Travis CI enviroment is done in tox.ini (this breaks tox runs on fedora, moreover this also means that Travis CI dev setup is described in 2 files instead of single one)
* some ubuntu packages installed in Travis CI dev setup are not available
* configuration of default tox environment disables build of sdist tarball, this should not be done in a default env like this, but only for tox test environments which doesn't need it
* test requirements should not be listed in test-requirements.txt file as it is now, because we have some unnecessary packages there and all those dependencies are installed for all test environments no matter if it makes sense or not - this in not manageable in the long run and it pollutes pip environment used for testing
* one should be able to run all tests via tox themselves (eg. developer on local machine with Fedora), which means that all details of how to run tests/checks should be done in tox.ini while travis.yml would contain only steps to setup and run tox in Travis Ubuntu machines
* split Travis CI run into multiple jobs, so that it's easier to review the test results
* unit tests directories should not contain any `__init__.py` files (it breaks pytest tox runs and it's against pytest good practices)
* Travis CI run should run sphinx build as well, to catch breakage of rst files in documentation


I'm going to provide a pull request with suggested fix for all issues listed there, with full explanation behind my suggested changes.

**Comment by mbukatov** [comment]
This issue has been fixed in just merged pull request #25. That said, tests are failing because of missing/incorrect dependencies - which is out of scope of this issue and has been reported as issue #17 and #18.
